R/check_frontmatter.R

```diff
@@ -8,103 +8,165 @@
+  }
+  author_name <- map_lgl(yml_protocol$author, ~is.string(.$name))
+  author_orcid <- map_lgl(yml_protocol$author, ~is.string(.$orcid))
+  problems <- c(problems,
+                sprintf(
+                  "Author name %s is not a string",
+                  map_chr(yml_protocol$author, "name")[!author_name]
+                ))
+  problems <- c(problems,
+                sprintf(
+                  "Author orcid %s is not a string",
+                  map_chr(yml_protocol$author, "orcid")[!author_orcid]
+                ))
+
+  names <- map_chr(yml_protocol$author, "name")
+  orcids <- map_chr(yml_protocol$author, "orcid")
+  problems <- c(problems,
+                "Number of author names not equal to number of orcids"[
+                  length(names) != length(orcids)])
+  problems <- c(problems,
+                "Please provide `orcids` in the `0000-0000-0000-0000` format."[
+                  !all(nchar(orcids) == 19)])
+  problems <- c(
+    problems,
+    "A single author should be passed as: c(\"lastname1, firstname1\")"[
+      !((is.string(names) & all(str_detect(names, ",{1}"))) |
+        is.character(names))])
+  problems <- c(
+    problems,
+    paste0("Multiple commas detected in author string.",
+           "Multiple authors should be passed as: ",
+           "c(\"lastname1, firstname1\", \"lastname2, firstname2\")")[
+             !((is.string(names) & !all(str_detect(names, ",{2,}"))) |
+               is.character(names))])
+  problems <- c(
+    problems,
+    "Multiple orcids should be passed as c(\"orcid1\", \"orcid2\")"[
+      any(str_detect(orcids, ",|;"))])
+
   if (!requireNamespace("lubridate", quietly = TRUE)) {
     stop("Package \"lubridate\" needed for checking of date. ",
          "Please install it with 'install.packages(\"lubridate\")'.",
          call. = FALSE)
   }
-  assert_that(isTRUE(
-    all.equal(yml$date,
-              lubridate::format_ISO8601(as.Date(yml$date)))
-  ),
-  msg = "'date' must be in YYYY-MM-DD format")
-
-  assert_that(has_name(yml, "reviewers"))
-  assert_that(is.character(yml$reviewers))
-
-  assert_that(has_name(yml, "file_manager"))
-  assert_that(is.string(yml$file_manager))
-
-  assert_that(has_name(yml, "protocol_code"))
-  assert_that(str_detect(yml$protocol_code, "^s[fioap]p-\\d{3}-(nl|en)$"))
-
-  assert_that(has_name(yml, "version_number"))
-  assert_that(str_detect(yml$version_number, "^\\d{4}\\.\\d{2}$"))
-
-  assert_that(has_name(yml, "language"))
-  assert_that(any(yml$language %in% c("nl", "en")),
-              is.string(yml$language),
-              msg = "'lang' must be 'nl' or 'en'")
-
-  if (has_name(yml, "theme")) {
-    assert_that(
-      any(yml$theme %in% c("generic", "water", "air", "soil", "vegetation",
-                              "species")),
-      is.string(yml$theme),
-      msg = paste0(
-        "Please check theme in yaml metadata\n",
-        "It should be one of generic, water, air, soil or vegetation")
-      )
+  if (
+    isFALSE(
```

Comment:
```suggestion
    !isTRUE(
```
`isFALSE()` gives FALSE both if `all.equal()` gives TRUE and if it gives for instance "1 string mismatch", so `isTRUE()` is needed to make the distinction between TRUE and another output.